core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
Then I think I might abandon this pull request so that this one can go back to the drawing board. It sounds like I don't really understand how #2491 should be fixed, either in the short term (work-around for instructions we can't decode) or in the longer term (how to handle UNPREDICTABLE instructions that are likely to read or write the stolen register or the PC).

Nevertheless, it seems wrong that a call to `instr_encode_arch` causes the _decoder_ to be invoked on the instruction that we are trying to _encode._